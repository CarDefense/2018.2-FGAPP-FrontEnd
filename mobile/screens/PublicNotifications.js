import { Constants, ImagePicker, Permissions } from 'expo';
import { NOTIFICATIONS_API } from './tab_navigator/car_defense/screens/TabNavigator/const/Const'
import React, { Component, Children } from 'react';
import {
  ActivityIndicator,
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  Alert,
  Image,
  KeyboardAvoidingView,
  Picker,
} from 'react-native';
import { Icon } from "native-base";
import { TextField } from 'react-native-material-textfield';


export default class PublicNotifications extends Component {

  constructor(props) {
    super(props);

    this.messageRef = this.updateRef.bind(this, 'message');
    this.onFocus = this.onFocus.bind(this);

    this.state = {
      title: 'Alerta Geral',
      message: '',
      image: null,
      uploading: false,
    }
  }
  updateTitle = (title) => {
    this.setState({ title: title })
  }

  onFocus() {
    let { errors = {} } = this.state;

    for (let text in errors) {
      let ref = this[text];

      if (ref && ref.isFocused()) {
        delete errors[text];
      }
    }

    this.setState({ errors });
  }

  onChangeText(words) {
    ['message']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
  }

  onPressButton = () => {
    let errors = {};
    const url = NOTIFICATIONS_API + '/send_emergency_push_message/' //function send_emergency_push_message url
    let i = true;
    const { state } = this.props.navigation;
    var id = state.params ? (state.params.user.id ? state.params.user.id : state.params.user.user_id) : undefined;

    ['message']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Campo obrigatório.';
          i += 1;
        } else {
          if ('message' === text && value.length < 5) {
            errors[text] = 'Forneça mais detalhes do ocorrido.';
            i += 1;
          }
        }
      });

    if (i) {
      let notification = JSON.stringify({
        sender_id: id,
        title: this.state.title,
        message: this.state.message,
        image: this.state.image
      })

      console.log(notification);

      fetch(url, {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: notification
      }).then(response => { return response.json() }
      ).then(jsonResponse => {
        console.log(jsonResponse);
      }
      ).catch(error => {
        console.log(error)
      })
      Alert.alert("Alerta enviado!")
    }

    this.setState({ errors });
  }

  updateRef(text, ref) {
    this[text] = ref;
  }

  render() {
    let { image } = this.state;
    let { errors = {}, ...data } = this.state;
    let { message = 'text' } = data;

    return (
      <View style={{ backgroundColor: '#00ACC1', flex: 1 }}>
        <KeyboardAvoidingView behavior="position">
          <ScrollView>
            <View style={styles.container}>
              <View style={styles.header}>
                <View style={styles.headerContent}>
                  <Image style={styles.avatar}
                    source={require('../images/alert.png')}
                  />
                </View>
              </View>
              <View style={styles.borderContainer}>
                <View style={styles.border}>
                  <Picker selectedValue={this.state.title} onValueChange={this.updateTitle} style={{ color: "#26C6DA", backgroundColor: 'white' }} mode="dropdown">
                    <Picker.Item label="Alerta Geral" value="Alerta Geral" />
                    <Picker.Item label="Roubo" value="Roubo" />
                    <Picker.Item label="Incêndio" value="Incendio" />
                    <Picker.Item label="Tempestade" value="Tempestade" />
                  </Picker>
                  <TextField
                    ref={this.messageRef}
                    value={data.message}
                    autoCorrect={false}
                    enablesReturnKeyAutomatically={true}
                    onFocus={this.onFocus}
                    onChangeText={(message) => this.setState({ message })}
                    maxLength={50}
                    returnKeyType='next'
                    label='Detalhes'
                    tintColor="white"
                    error={errors.message}
                    textColor="white"
                    labelPadding={5}
                    placeholderTextColor="white"
                    inputContainerStyle={{ marginHorizontal: 20 }}
                  />
                  <View style={styles.alternativeLayoutButtonContainer}>
                    <TouchableOpacity
                      color="#B2EBF2"
                      onPress={this._takePhoto}
                      containerViewStyle={{ width: '10%' }}
                    >
                      <Icon
                        type='FontAwesome'
                        name="camera"
                        style={{ color: "#26C6DA" }}
                      />
                    </TouchableOpacity>
                    <TouchableOpacity
                      color="#B2EBF2"
                      onPress={this._pickImage}
                      containerViewStyle={{ width: '10%' }}
                    >
                      <Icon
                        type='FontAwesome'
                        name="image"
                        style={{ color: "#26C6DA" }}
                      />
                    </TouchableOpacity>
                  </View>
                  {this._maybeRenderImage()}
                  {this._maybeRenderUploadingOverlay()}
                  <View style={styles.container1}>
                    <TouchableOpacity
                      style={styles.button3}
                      backgroundColor="#26C6DA"
                      onPress={this.onPressButton}
                      containerViewStyle={{ width: '30%' }}
                    >
                      <Text style={{ color: '#26C6DA', fontWeight: '800' }} >Enviar</Text>
                    </TouchableOpacity>
                  </View>
                </View>
              </View>
            </View>
          </ScrollView>
        </KeyboardAvoidingView>
      </View>
    );
  }

  _maybeRenderUploadingOverlay = () => { //done
    if (this.state.uploading) {
      return (
        <View
          style={styles.maybeRenderUploading}>
          <ActivityIndicator color="#540b71" size="large" />
        </View>
      );
    }
  };

  _maybeRenderImage = () => { //done
    let {
      image
    } = this.state;

    if (!image) {
      return;
    }

    // onPress={this._copyToClipboard}
    // onLongPress={this._share}

    return (
      <View
        style={styles.maybeRenderContainer}>
        <View
          style={styles.maybeRenderImageContainer}>
          <Image source={{ uri: image }} style={styles.maybeRenderImage} />
        </View>
      </View>
    );
  };

  _share = () => {
    Share.share({
      message: this.state.image,
      title: 'Check out this photo',
      url: this.state.image,
    });
  };

  _copyToClipboard = () => {
    Clipboard.setString(this.state.image);
    alert('Copied image URL to clipboard');
  };

  _takePhoto = async () => {
    const {
      status: cameraPerm
    } = await Permissions.askAsync(Permissions.CAMERA);

    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera AND camera roll
    if (cameraPerm === 'granted' && cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchCameraAsync({
        allowsEditing: true,
      });

      this._handleImagePicked(pickerResult);
    }
  };

  _pickImage = async () => {
    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera roll
    if (cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchImageLibraryAsync({
        allowsEditing: true,

      });

      this._handleImagePicked(pickerResult);
    }
  };

  _handleImagePicked = async pickerResult => {
    let uploadResponse, uploadResult;

    try {
      this.setState({
        uploading: true
      });

      if (!pickerResult.cancelled) {
        uploadResponse = await uploadImageAsync(pickerResult.uri);
        uploadResult = await uploadResponse.json();

        console.log(uploadResult.image)

        this.setState({
          image: uploadResult.image
        });
      }
    } catch (e) {
      console.log({ uploadResponse });
      console.log({ uploadResult });
      console.log({ e });
      alert('Falha de Upload, tente novamente.');
    } finally {
      this.setState({
        uploading: false
      });
    }
  };
}

async function uploadImageAsync(uri) {
  let apiUrl = NOTIFICATIONS_API + '/notificationsimage/';

  // Note:
  // Uncomment this if you want to experiment with local server
  //
  // if (Constants.isDevice) {
  //   apiUrl = `https://your-ngrok-subdomain.ngrok.io/upload`;
  // } else {
  //   apiUrl = `http://localhost:3000/upload`
  // }

  let uriParts = uri.split('.');
  let fileType = uriParts[uriParts.length - 1];

  let formData = new FormData();
  formData.append('image', {
    uri,
    name: `photo.${fileType}`,
    type: `image/${fileType}`,
  });

  console.log(uri)

  let options = {
    method: 'POST',
    body: formData,
    headers: {
      Accept: 'application/json',
      'Content-Type': 'multipart/form-data',
    },
  };
  return fetch(apiUrl, options);
}

const styles = StyleSheet.create({
  container: {
    paddingLeft: 16,
    paddingRight: 16,
    flex: 1
  },
  text: {
    paddingTop: 16,
    color: "#B2EBF2",
    fontWeight: 'bold',
  },
  alternativeLayoutButtonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    paddingTop: 4,
    paddingRight: 1,
    marginTop: 20
  },
  button: {
    backgroundColor: "white",
    borderRadius: 15,
    height: 30,
    width: 190,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  button2: {
    backgroundColor: "white",
    borderRadius: 15,
    height: 40,
    width: 120,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  button3: {
    backgroundColor: "white",
    borderRadius: 15,
    height: 40,
    width: 120,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  container1: {
    paddingTop: 20,
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: 40
  },
  header: {
    marginTop: 25,
  },
  maybeRenderUploading: {
    paddingTop: 8,
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.0)',
    justifyContent: 'center',
  },
  maybeRenderContainer: {
    width: '100%',
    height: '41%',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3,
    elevation: 2,
    marginTop: 10,
    shadowColor: 'rgba(0,0,0,1)',
    shadowOpacity: 0.2,
    shadowOffset: {
      height: 4,
      width: 4,
    },
    shadowRadius: 5
  },
  maybeRenderImageContainer: {
    borderTopLeftRadius: 3,
    borderTopRightRadius: 3,
    overflow: 'hidden',
  },
  maybeRenderImage: {
    height: 200,
    width: 310,
  },
  avatar: {
    width: 100,
    height: 100,
    borderRadius: 40,
    borderWidth: 5,
    borderColor: "white",
    marginBottom: 10,
  },
  name: {
    fontSize: 18,
    color: "white",
    fontWeight: '800',
  },
  headerContent: {
    padding: 30,
    alignItems: 'center',
  },
  input: {
    marginBottom: 5,
    paddingLeft: 10,
    paddingRight: 10
  },
  border: {
    borderWidth: 5,
    borderRadius: 15,
    borderColor: "white",
    backgroundColor: '#B2EBF2'
  },
  borderContainer: {
    paddingBottom: 20,
    marginBottom: 20
  }
});
