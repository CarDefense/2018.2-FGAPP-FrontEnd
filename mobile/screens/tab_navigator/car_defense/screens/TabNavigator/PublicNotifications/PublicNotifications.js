import { TextField } from 'react-native-material-textfield';;
import { Constants, ImagePicker, Permissions } from 'expo';
import { NOTIFICATIONS_API } from '../const/Const'
import React, { Component, Children } from 'react';
import {
  ActivityIndicator,
  View,
  Text,
  StyleSheet,
  ScrollView,
  TouchableOpacity,
  Alert,
  Image,
  KeyboardAvoidingView,
  Picker,
} from 'react-native';
import { Icon } from "native-base";


const tk = 0;

export default class PublicNotifications extends Component {

  constructor(props) {
    super(props);

    this.messageRef = this.updateRef.bind(this, 'message');
    this.onFocus = this.onFocus.bind(this);

    this.state = {
      title: '',
      message: '',
      image: null,
      uploading: false,
    }
  }
  updateTitle = (title) => {
    this.setState({ title: title})
    }
    
  onFocus() {
    let { errors = {} } = this.state;

    for (let text in errors) {
      let ref = this[text];

      if (ref && ref.isFocused()) {
        delete errors[text];
      }
    }

    this.setState({ errors });
  }

  onChangeText(words) {
    ['message']
      .map((text) => ({ text, ref: this[text] }))
      .forEach(({ text, ref }) => {
        if (ref && ref.isFocused()) {
          this.setState({ [text]: words });
        }
      });
  }

  // onSubmitFirstName() {
  //   this.lastname.focus();
  // }

  onPressButton = () => {
    let errors = {};
    const url = NOTIFICATIONS_API + '/send_emergency_push_message/' //function send_emergency_push_message url
    let i = 0;

    ['message']
      .forEach((text) => {
        let value = this[text].value();

        if (!value) {
          errors[text] = 'Campo obrigatório.';
          i += 1;
        } else {
          if ('message' === text && value.length < 5) {
            errors[text] = 'Forneça mais detalhes do ocorrido.';
            i += 1;
          }
        }
      });

    if (i == 0) {
      let notification = JSON.stringify({
        sender_id: tk,
        title: this.state.title,
        message: this.state.message,
        image: this.state.image
      })

      console.log(notification);

      fetch(url, {
        method: 'POST',
        headers: {
          'Accept': 'application/json',
          'Content-Type': 'application/json',
        },
        body: notification
      }).then(response => { return response.json() }
      ).then(jsonResponse => {
        console.log(jsonResponse);
      }
      ).catch(error => {
        console.log(error)
      })
      Alert.alert("Alerta enviado!")
    }

    this.setState({ errors });
  }

  updateRef(text, ref) {
    this[text] = ref;
  }

  render() {
    let { image } = this.state;
    let { errors = {}, ...data } = this.state;
    let { message = 'text' } = data;

    return (
      <View style={{ backgroundColor: '#8bd4da', flex: 1 }}>
      <KeyboardAvoidingView behavior="position">
        <ScrollView>
          <View style={styles.container}>
            <View style={styles.header}>
              <View style={styles.headerContent}>
                <Image style={styles.avatar}
                  source={require('../../../../../../images/alert.png')}
                />
                <Text style={styles.name}>Envie aqui os alertas</Text>
              </View>
            </View>
            <Picker selectedValue= {this.state.title} onValueChange={this.updateTitle} style={{ color: "white" }} mode="dropdown">           
              <Picker.Item label="Alerta Geral" value="Alerta Geral" />
              <Picker.Item label="Roubo" value="Roubo" />
              <Picker.Item label="Incêndio" value="Incendio" />
              <Picker.Item label ="Tempestade" value="Tempestade" />
            </Picker>           
            <TextField
              ref={this.messageRef}
              value={data.message}
              autoCorrect={false}
              enablesReturnKeyAutomatically={true}
              onFocus={this.onFocus}
              onChangeText={(message) => this.setState({ message })}
              // onChangeText={this.onChangeText}
              // onSubmitEditing={this.onSubmitMessage}
              returnKeyType='next'
              label='Detalhes'
              tintColor="white"
              error={errors.message}
              textColor="white"
              labelPadding={10}
              placeholderTextColor="white"
            />
            <View style={styles.alternativeLayoutButtonContainer}>
              <TouchableOpacity
                color="#8bd4da"
                onPress={this._takePhoto}
                containerViewStyle={{ width: '10%' }}
              >
                <Icon
                  type='FontAwesome'
                  name="camera"
                  style={{ color: "white" }}
                />
              </TouchableOpacity>
              <TouchableOpacity
                color="#8bd4da"
                onPress={this._pickImage}
                containerViewStyle={{ width: '10%' }}
              >
                <Icon
                  type='FontAwesome'
                  name="image"
                  style={{ color: "white" }}
                />
              </TouchableOpacity>
            </View>
            {this._maybeRenderImage()}
            {this._maybeRenderUploadingOverlay()}
            <View style={styles.container1}>
              <TouchableOpacity
                style={styles.button3}
                color="#8bd4da"
                onPress={this.onPressButton}
                containerViewStyle={{ width: '40%' }}
              >
                <Text style={{ color: '#8bd4da', fontWeight: '800' }} >Enviar</Text>
              </TouchableOpacity>
            </View>
          </View>
        </ScrollView>
      </KeyboardAvoidingView>
      </View>
    );
  }

  _maybeRenderUploadingOverlay = () => { //done
    if (this.state.uploading) {
      return (
        <View
          style={styles.maybeRenderUploading}>
          <ActivityIndicator color="#540b71" size="large" />
        </View>
      );
    }
  };

  _maybeRenderImage = () => { //done
    let {
      image
    } = this.state;

    if (!image) {
      return;
    }

    // onPress={this._copyToClipboard}
    // onLongPress={this._share}

    return (
      <View
        style={styles.maybeRenderContainer}>
        <View
          style={styles.maybeRenderImageContainer}>
          <Image source={{ uri: image }} style={styles.maybeRenderImage} />
        </View>
      </View>
    );
  };

  _share = () => {
    Share.share({
      message: this.state.image,
      title: 'Check out this photo',
      url: this.state.image,
    });
  };

  _copyToClipboard = () => {
    Clipboard.setString(this.state.image);
    alert('Copied image URL to clipboard');
  };

  _takePhoto = async () => {
    const {
      status: cameraPerm
    } = await Permissions.askAsync(Permissions.CAMERA);

    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera AND camera roll
    if (cameraPerm === 'granted' && cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchCameraAsync({
        allowsEditing: true,
      });

      this._handleImagePicked(pickerResult);
    }
  };

  _pickImage = async () => {
    const {
      status: cameraRollPerm
    } = await Permissions.askAsync(Permissions.CAMERA_ROLL);

    // only if user allows permission to camera roll
    if (cameraRollPerm === 'granted') {
      let pickerResult = await ImagePicker.launchImageLibraryAsync({
        allowsEditing: true,

      });

      this._handleImagePicked(pickerResult);
    }
  };

  _handleImagePicked = async pickerResult => {
    let uploadResponse, uploadResult;

    try {
      this.setState({
        uploading: true
      });

      if (!pickerResult.cancelled) {
        uploadResponse = await uploadImageAsync(pickerResult.uri);
        uploadResult = await uploadResponse.json();

        console.log(uploadResult.image)

        this.setState({
          image: uploadResult.image
        });
      }
    } catch (e) {
      console.log({ uploadResponse });
      console.log({ uploadResult });
      console.log({ e });
      alert('Upload failed, sorry :(');
    } finally {
      this.setState({
        uploading: false
      });
    }
  };
}

async function uploadImageAsync(uri) {
  let apiUrl = NOTIFICATIONS_API + '/notificationsimage/';

  // Note:
  // Uncomment this if you want to experiment with local server
  //
  // if (Constants.isDevice) {
  //   apiUrl = `https://your-ngrok-subdomain.ngrok.io/upload`;
  // } else {
  //   apiUrl = `http://localhost:3000/upload`
  // }

  let uriParts = uri.split('.');
  let fileType = uriParts[uriParts.length - 1];

  let formData = new FormData();
  formData.append('image', {
    uri,
    name: `photo.${fileType}`,
    type: `image/${fileType}`,
  });

  console.log(uri)

  let options = {
    method: 'POST',
    body: formData,
    headers: {
      Accept: 'application/json',
      'Content-Type': 'multipart/form-data',
    },
  };
  return fetch(apiUrl, options);
}

const styles = StyleSheet.create({
  container: {
    paddingLeft: 16,
    paddingRight: 16,
    backgroundColor: '#8bd4da',
    flex: 1
  },
  text: {
    paddingTop: 16,
    color: "#8bd4da",
    fontWeight: 'bold',
  },
  alternativeLayoutButtonContainer: {
    flexDirection: 'row',
    justifyContent: 'space-evenly',
    paddingTop: 4,
    paddingRight: 1,
    marginTop: 20
  },
  button: {
    backgroundColor: "white",
    borderRadius: 15,
    height: 30,
    width: 190,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  button2: {
    backgroundColor: "white",
    borderRadius: 15,
    height: 40,
    width: 120,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  button3: {
    backgroundColor: "white",
    borderRadius: 15,
    height: 40,
    width: 120,
    flexDirection: 'row',
    justifyContent: 'center',
    alignItems: 'center'
  },
  container1: {
    paddingTop: 20,
    flexDirection: 'row',
    justifyContent: 'center',
    marginBottom: 40
  },
  header: {
    marginTop: 25,
  },
  maybeRenderUploading: {
    paddingTop: 8,
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.0)',
    justifyContent: 'center',
  },
  maybeRenderContainer: {
    width: '100%',
    height: '40%',
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 3,
    elevation: 2,
    marginTop: 32,
    shadowColor: 'rgba(0,0,0,1)',
    shadowOpacity: 0.2,
    shadowOffset: {
      height: 4,
      width: 4,
    },
    shadowRadius: 5
  },
  maybeRenderImageContainer: {
    borderTopLeftRadius: 3,
    borderTopRightRadius: 3,
    overflow: 'hidden',
  },
  maybeRenderImage: {
    height: 250,
    width: 250,
  },
  avatar: {
    width: 170,
    height: 170,
    borderRadius: 63,
    borderWidth: 7,
    borderColor: "white",
    marginBottom: 10,
  },
  name: {
    fontSize: 18,
    color: "white",
    fontWeight: '800',
  },
  headerContent: {
    padding: 30,
    alignItems: 'center',
  },
  input: {
    marginBottom: 5,
    paddingLeft: 10,
    paddingRight: 10
  }
});
