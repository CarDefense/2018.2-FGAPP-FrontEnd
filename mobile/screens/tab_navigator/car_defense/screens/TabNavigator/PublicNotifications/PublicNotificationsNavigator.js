import React, { Component } from "react";
import {StackNavigator} from 'react-navigation'

import PublicNotifications from './PublicNotifications'


export default class PublicNotificationsNavigator extends Component {

    render() {
        return (
            <PublicNotificationsStackNavigator />
        );
    }
}

const PublicNotificationsStackNavigator = new StackNavigator({

    PublicNotificatios: {
        screen: PublicNotifications,
        navigationOptions: ({ navigation }) => ({
            header: null,
        }),
    },
})
