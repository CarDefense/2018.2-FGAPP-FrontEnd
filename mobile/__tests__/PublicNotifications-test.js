import PublicNotifications from '../screens/tab_navigator/car_defense/screens/TabNavigator/PublicNotifications/PublicNotifications'
import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

test('PublicNotifications snapShot', () => {
    const snap = renderer.create(
        <PublicNotifications/>
    ).toJSON()
    expect(snap).toMatchSnapshot()
})