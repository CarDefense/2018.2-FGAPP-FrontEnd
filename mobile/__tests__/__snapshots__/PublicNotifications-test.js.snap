// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicNotifications snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ACC1",
      "flex": 1,
    }
  }
>
  <View
    enabled={true}
    keyboardVerticalOffset={0}
    onLayout={[Function]}
    style={undefined}
  >
    <View
      style={
        Array [
          undefined,
          Object {
            "bottom": 0,
          },
        ]
      }
    >
      <RCTScrollView>
        <View>
          <View
            style={
              Object {
                "flex": 1,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
          >
            <View
              style={
                Object {
                  "marginTop": 25,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "padding": 30,
                  }
                }
              >
                <Image
                  source={1}
                  style={
                    Object {
                      "borderColor": "white",
                      "borderRadius": 40,
                      "borderWidth": 5,
                      "height": 100,
                      "marginBottom": 10,
                      "width": 100,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "marginBottom": 20,
                  "paddingBottom": 20,
                }
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#B2EBF2",
                    "borderColor": "white",
                    "borderRadius": 15,
                    "borderWidth": 5,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "backgroundColor": "white",
                      "color": "#26C6DA",
                    }
                  }
                >
                  <RCTPicker
                    items={
                      Array [
                        Object {
                          "label": "Alerta Geral",
                          "textColor": undefined,
                          "value": "Alerta Geral",
                        },
                        Object {
                          "label": "Roubo",
                          "textColor": undefined,
                          "value": "Roubo",
                        },
                        Object {
                          "label": "IncÃªndio",
                          "textColor": undefined,
                          "value": "Incendio",
                        },
                        Object {
                          "label": "Tempestade",
                          "textColor": undefined,
                          "value": "Tempestade",
                        },
                      ]
                    }
                    onChange={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    selectedIndex={0}
                    style={
                      Array [
                        Object {
                          "height": 216,
                        },
                        undefined,
                      ]
                    }
                  />
                </View>
                <View
                  onResponderRelease={[Function]}
                  onStartShouldSetResponder={[Function]}
                  pointerEvents="auto"
                  style={undefined}
                >
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "borderBottomColor": "rgba(0, 0, 0, 0.38)",
                        "borderBottomWidth": 0.5,
                        "height": 64,
                        "marginHorizontal": 20,
                        "paddingBottom": 8,
                        "paddingTop": 32,
                      }
                    }
                  >
                    <View
                      collapsable={undefined}
                      style={
                        Object {
                          "position": "absolute",
                          "top": 36,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        collapsable={undefined}
                        ellipsizeMode="tail"
                        numberOfLines={1}
                        style={
                          Object {
                            "color": "rgba(0, 0, 0, 0.38)",
                            "fontSize": 16,
                          }
                        }
                      >
                        Detalhes
                      </Text>
                    </View>
                    <View
                      style={
                        Object {
                          "flexDirection": "row",
                        }
                      }
                    >
                      <TextInput
                        allowFontScaling={true}
                        autoCapitalize="sentences"
                        autoCorrect={false}
                        disableFullscreenUI={true}
                        editable={true}
                        enablesReturnKeyAutomatically={true}
                        error={undefined}
                        maxLength={50}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onChangeText={[Function]}
                        onContentSizeChange={[Function]}
                        onFocus={[Function]}
                        placeholderTextColor="white"
                        returnKeyType="next"
                        selectionColor="white"
                        style={
                          Array [
                            Object {
                              "flex": 1,
                              "margin": 0,
                              "padding": 0,
                              "top": 2,
                            },
                            Object {
                              "color": "white",
                              "fontSize": 16,
                              "height": 24,
                              "textAlign": "left",
                            },
                            undefined,
                          ]
                        }
                        underlineColorAndroid="transparent"
                        value=""
                      />
                    </View>
                  </View>
                  <View
                    collapsable={undefined}
                    style={
                      Object {
                        "flexDirection": "row",
                        "height": 8,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "bottom": 0,
                            "left": 0,
                            "paddingVertical": 4,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          collapsable={undefined}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "color": "rgb(213, 0, 0)",
                              "fontSize": 0,
                              "opacity": 0,
                            }
                          }
                        />
                      </View>
                      <View
                        style={
                          Object {
                            "alignItems": "flex-start",
                            "bottom": 0,
                            "left": 0,
                            "paddingVertical": 4,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={true}
                          collapsable={undefined}
                          ellipsizeMode="tail"
                          numberOfLines={1}
                          style={
                            Object {
                              "backgroundColor": "transparent",
                              "color": "rgba(0, 0, 0, .38)",
                              "fontSize": 12,
                              "opacity": 1,
                            }
                          }
                        />
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-evenly",
                      "marginTop": 20,
                      "paddingRight": 1,
                      "paddingTop": 4,
                    }
                  }
                >
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    collapsable={undefined}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    collapsable={undefined}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                    />
                  </View>
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginBottom": 40,
                      "paddingTop": 20,
                    }
                  }
                >
                  <View
                    accessibilityComponentType={undefined}
                    accessibilityLabel={undefined}
                    accessibilityTraits={undefined}
                    accessible={true}
                    collapsable={undefined}
                    hasTVPreferredFocus={undefined}
                    hitSlop={undefined}
                    isTVSelectable={true}
                    nativeID={undefined}
                    onLayout={undefined}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "white",
                        "borderRadius": 15,
                        "flexDirection": "row",
                        "height": 40,
                        "justifyContent": "center",
                        "opacity": 1,
                        "width": 120,
                      }
                    }
                    testID={undefined}
                    tvParallaxProperties={undefined}
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "#26C6DA",
                          "fontWeight": "800",
                        }
                      }
                    >
                      Enviar
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
    </View>
  </View>
</View>
`;
